### DISCUSSION TITLE
Some questions about the integration to the alarm system, Gen2

### DISCUSSION THREAD

**Discussion by gurkburk76** [tsightler/ring-mqtt#657] [General Discussion]
I hope i'm not too far out of scope, but i tried to look for answers for a few questions that popped up when looking a a decent home alarm system that is not dependent on the cloud, the name ring leaves a bit of a bad taste in my mouth, at least historically when it comes to doorbells but this system seems cost effective and not bad judging from reviews.

What i mostly need to know is if it's a local-only implementation and how it works, wifi/zwave etc, i'm assuming the alarm system uses wifi to connect to the world, it  also seems to operate via zwave with it's devices ?
Can i do something in home assistant to integrate it that way or does the main unit spit out stuff via mqtt / via wifi ?

I think we all know that manufacturers can kill off cloud dependant stuff when ever they feel like it so it would be REALLY good to be able to control it via HA when that happens.

Thanks!

**Comment by tsightler** [comment]
I'm going to move this to the discussions sections as this is not a support issue.

Also, you question is largely already answered [in the FAQ](https://github.com/tsightler/ring-mqtt/wiki/Frequently-Asked-Questions-(FAQ)), third question down.

If you want slightly more detail:

1) Ring alarm base station is basically a Z-wave hub that maintains a websocket connection to Ring over which is sends all sensor updates and state to Ring servers (that's how the Ring apps get status).
2) Ring alarm base station is "smart", in that it connects to the sensors via Z-wave and can be put into arm/disarm modes locally via the keypad, however, this provides very limited functionality and there's no way to control the base station locally, as all communications goes through Ring servers.  If internet connectivity is down, there is limited backup connectivity via cell service, so the alarm will still trigger if a local sensor trips and central monitoring can be alerted, but all interactive features do not work in this case.
3) Ring alarm sensors are mostly just Z-wave devices and it's generally possible to use them with any Z-wave compatible hubs.  They are generally good quality sensors, but there are cheaper alternatives that also work fine if you are building your own.

Summary, Ring alarm, like all Ring products, are cloud based.  If you looking for 100% local control, look elsewhere.

Realistically, if you just want to build a local alarm, with no remote monitoring, you don't need anything but HA, some Z-wave/Zigbee/<insert favorite sensor protocol> sensors and a USB stick to support them, and something like the [Alarmo integration](https://github.com/nielsfaber/alarmo).  Alarmo allows you to use any sensors and devices connected to HA to build a very flexible, local alarm system.

Of course, if you are looking for professional monitoring, that will be an issue with a homebrew, local-only system.  For me, alarm monitoring is a critical component, so Ring alarm being cloud controlled was a tradeoff I was willing to accept (even my 20 year old ADT alarm was connected via the Internet and had a Web control panel I had to log into, so it wasn't that much different).

**Reply by gurkburk76** [reply]
Oh, apologies, i totally missed that there is a discussions section :)
I'm reasonably "fine" with it using the cloud, but i mainly wanted to guard against creating more e-waste when ring decides that it's not gonna support the product any longer.
Judging from what i'm hearing when that time comes, and assuming the sensors work as normal gadget in a zwave network and/or have the abilitiy to work with another controller it should be resonable easy to reuse them i think. Basically make my own alarmo solution with it, should be fine.

Today i own a envisalink + dsc pc1616 that is starting to behave ..strange so i'm just looking for solutions when the day i need them comes.

**Reply by tsightler** [reply]
There are quite a few pages out there describing how to use Ring sensors with generic Z-wave devices.  I've personally only tried the contact sensors, but they worked easily enough.  There are even pages that go into details about the keypad and how to make it do things with HA directly so most of the devices are somewhat salvageable.

Ring seems to have a pretty good commitment to hardware so far.  They've already committed to software updates until at least 2027 on all of their devices (many of those devices will be 10+ years old by then, my alarm and cameras will be 9 years old at that point).  Many companies that kill off old hardware do so because they are not getting revenue from them, but Ring gets lots of recurring revenue from their hardware via subscriptions so they probably won't kill it off too quickly, probably not until it costs more than warranty claims against said hardware.  But that is just a guess on my part.

Then again, they might kill off the old alarm 1st gen alarm system after 2027 because they might be able to use it as an excuse to sunset the old Protect Plans that I'm grandfathered into.  $100/yr for professional alarm monitoring plus unlimited recording for cameras is basically a steal.  At $200/yr, I'll be less likely to renew, and I'll probably go with local cameras the next time around as they have improved dramatically in usability, capability and quality since then.
